go/genesis/genesis_test.go

```diff
@@ -695,39 +709,40 @@ func TestGenesisSanityCheck(t *testing.T) {
 
 	// NOTE: There doesn't seem to be a way to generate invalid Quantities, so
 	// we're just going to test the code that checks if things add up.
-	d = *testDoc
+	d = testDoc()
 	d.Staking.TotalSupply = *quantity.NewFromUint64(100)
 	require.Error(d.SanityCheck(), "invalid total supply should be rejected")
 
-	d = *testDoc
+	d = testDoc()
 	d.Staking.CommonPool = *quantity.NewFromUint64(100)
 	require.Error(d.SanityCheck(), "invalid common pool should be rejected")
 
-	d = *testDoc
+	d = testDoc()
 	d.Staking.LastBlockFees = *quantity.NewFromUint64(100)
 	require.Error(d.SanityCheck(), "invalid last block fees should be rejected")
 
-	d = *testDoc
+	d = testDoc()
 	d.Staking.Ledger[stakingTests.DebugStateSrcAddress].General.Balance = *quantity.NewFromUint64(100)
 	require.Error(d.SanityCheck(), "invalid general balance should be rejected")
 
-	d = *testDoc
-	d.Staking.Ledger[stakingTests.DebugStateSrcAddress].Escrow.Active.Balance = *quantity.NewFromUint64(100)
-	require.Error(d.SanityCheck(), "invalid escrow active balance should be rejected")
+	// TODO: this doesn't actually fail, but it failed before because above mutation was propagated.
```

Comment:
Hm why doesn't this fail, is there a bug in the sanity checker?

Reply:
looked into this now, it was just that the `100` escrow balance matched the initial (correct) genesis state :-) fixed by setting a different value.